### PR TITLE
Add recursive option

### PR DESCRIPTION
**Pull request by phmarti**
The recursive option '-r' will process all '.f90', '.inc' and '.incf' (case insensitive) files in directories given as argument and build the same directory tree in output-dir with the preprocessed files.

### CODE REVIEW

File: src/serialbox-python/pp_ser/pp_ser.py [hunk 1]
```diff
@@ -54,6 +54,21 @@ def to_ascii(text):
     else:
         return str(text)
 
+def filter_fortran(f):
+    return (f.split('.')[-1].lower() in ['f90','inc','incf'])
```
Comment (clementval) [inline src/serialbox-python/pp_ser/pp_ser.py]: what about `.f`, `.f03`

File: src/serialbox-python/pp_ser/pp_ser.py [hunk 2]
```diff
@@ -54,6 +54,21 @@ def to_ascii(text):
     else:
         return str(text)
 
+def filter_fortran(f):
```
Comment (clementval) [inline src/serialbox-python/pp_ser/pp_ser.py]: one more blank line for PEP8 coding style

File: src/serialbox-python/pp_ser/pp_ser.py [hunk 3]
```diff
@@ -54,6 +54,21 @@ def to_ascii(text):
...
+        if not os.path.isdir(dest):
+            os.makedirs(dest)
+        files = os.listdir(src)
+        filtered_list.extend([os.path.join(src,f) for f in files if os.path.isfile(os.path.join(src,f)) and file_filter(f)])
+        dirs = [f for f in files if os.path.isdir(os.path.join(src,f))]
+        for d in dirs:
+            build_tree(os.path.join(src, d), os.path.join(dest, d), filtered_list, file_filter)
+    else:
+        if os.path.isfile(src) and file_filter(src):
+            filtered_list.append(src)
```
Comment (clementval) [inline src/serialbox-python/pp_ser/pp_ser.py]: one more blank line for PEP8 coding style

**Comment by havogt** [comment]
launch cosmo

**Pull request merged**